src/subject_name/verify.rs

```diff
@@ -320,35 +317,23 @@ impl<'a> Iterator for NameIterator<'a> {
 #[cfg(feature = "alloc")]
 pub(crate) fn list_cert_dns_names<'names>(
     cert: &'names crate::EndEntityCert<'names>,
-) -> Result<impl Iterator<Item = GeneralDnsNameRef<'names>>, Error> {
+) -> impl Iterator<Item = GeneralDnsNameRef<'names>> {
```

Comment:
Do you think these changes would be easier to review if you use two commits for (a) making this infallible and (b) getting rid of the allocation?

Reply:
It didn't feel like it would when I wrote it, but it sounds like as a reviewer you think it would :laughing: I can split it.

Reply:
Split this into two parts.